File: frontend/src/api/service.ts
import http from './_http';

export default {
  index (workspaceId: number) {
    return http.get(`/workspaces/${workspaceId}/services/`);
  },
  create (workspaceId: number, data: any) {
    return http.post(`/workspaces/${workspaceId}/services/`, data);
  },
  update (workspaceId: number, id: number, data: any) {
    return http.put(`/workspaces/${workspaceId}/services/${id}`, data);
  },
  delete (workspaceId: number, id: number) {
    return http.delete(`/workspaces/${workspaceId}/services/${id}`);
  },
  externalIndex (workspaceId: number) {
    return http.get(`/workspaces/${workspaceId}/services/external`)
  }
}

File: backend/routes/service.go
package routes

import (
	"app/controllers/service"
	"app/middlewares"

	"github.com/gin-gonic/gin"
)

func InitService(r *gin.RouterGroup, e *gin.Engine) {
	const baseUrl = "/:id/services"

	r.GET(baseUrl + "/", middlewares.CheckWorkspace, service.Index)
	r.POST(baseUrl + "/", middlewares.CheckWorkspace, service.Create)
	r.GET(baseUrl + "/external", service.ExternalIndex)
	r.PUT(baseUrl + "/:serviceId", middlewares.CheckWorkspace, service.Update)
	r.DELETE(baseUrl + "/:serviceId", middlewares.CheckWorkspace, service.Delete)
}
